internal/factory/marketDescription.go

```diff
@@ -3,23 +3,46 @@ package factory
 import (
 	"github.com/oddin-gg/gosdk/internal/cache"
 	"github.com/oddin-gg/gosdk/protocols"
+	"github.com/pkg/errors"
 )
 
 // MarketDescriptionFactory ...
 type MarketDescriptionFactory struct {
 	marketDescriptionCache *cache.MarketDescriptionCache
 	marketVoidReasonsCache *cache.MarketVoidReasonsCache
+	playerCache            *cache.PlayersCache
 }
 
-// MarketDescriptionByID ...
-func (m MarketDescriptionFactory) MarketDescriptionByID(marketID uint, specifiers map[string]string, locales []protocols.Locale) protocols.MarketDescription {
+// MarketDescriptionByIdAndSpecifiers ...
+func (m MarketDescriptionFactory) MarketDescriptionByIdAndSpecifiers(
+	marketID uint,
+	specifiers map[string]string,
+	locales []protocols.Locale,
+) (protocols.MarketDescription, error) {
 	var variant *string
 	specifier, ok := specifiers["variant"]
 	if ok {
 		variant = &specifier
 	}
 
-	return cache.NewMarketDescription(marketID, variant, m.marketDescriptionCache, locales)
+	return m.MarketDescriptionByIdAndVariant(marketID, variant, locales)
+}
+
+// MarketDescriptionByIdAndVariant ...
```

Comment:
**[BETTER-UX]:** IMHO, public API would deserve at least brief but reasonable comment (summary). Nice-to-have.